src/test/app/SHAMapStore_test.cpp

```diff
@@ -517,12 +521,96 @@ class SHAMapStore_test : public beast::unit_test::suite
         lastRotated = ledgerSeq - 1;
     }
 
+    static std::unique_ptr<Config>
+    rwdbNoDeleteDefaultConfig()
+    {
+        // RWDB without online_delete, in standalone mode (default)
+        auto cfg = test::jtx::envconfig();
+        cfg->section(ConfigSection::nodeDatabase()).set("type", "rwdb");
+        cfg->section(ConfigSection::nodeDatabase()).set("path", "main");
+        // Default is standalone = true from envconfig.cpp
+        return cfg;
+    }
+
+    static std::unique_ptr<Config>
+    rwdbNoDeleteNotStandalone()
+    {
+        // RWDB without online_delete and NOT in standalone mode
+        // This should trigger the enforcement
+        auto cfg = test::jtx::envconfig();
+        cfg->section(ConfigSection::nodeDatabase()).set("type", "rwdb");
+        cfg->section(ConfigSection::nodeDatabase()).set("path", "main");
+        cfg->setupControl(
+            true, true, false);  // bQuiet, bSilent, bStandalone=false
+        return cfg;
+    }
+
+    static std::unique_ptr<Config>
+    rwdbWithDeleteNotStandalone()
+    {
+        // RWDB with online_delete and NOT in standalone mode
+        auto cfg = test::jtx::envconfig();
+        cfg->section(ConfigSection::nodeDatabase()).set("type", "rwdb");
+        cfg->section(ConfigSection::nodeDatabase()).set("path", "main");
+        cfg->section(ConfigSection::nodeDatabase()).set("online_delete", "256");
+        cfg->setupControl(
+            true, true, false);  // bQuiet, bSilent, bStandalone=false
+        return cfg;
+    }
+
+    void
+    testRWDBOnlineDeleteEnforcement()
+    {
+        testcase("RWDB online_delete enforcement");
```

Comment:
I don't really like this here, but it is what it is 
It's testing code in SHAMapStoreImp.cpp, which had preexisting validation code that throwed exceptions
Take the L and move on?